packages/babel-plugin-transform-modules-commonjs/src/index.js

```diff
@@ -16,9 +16,19 @@ export default function(api, options) {
     strict,
     strictMode,
     noInterop,
+    lazy = false,
     // Defaulting to 'true' for now. May change before 7.x major.
     allowCommonJSExports = true,
   } = options;
+
+  if (
+    typeof lazy !== "boolean" &&
+    typeof lazy !== "function" &&
+    (!Array.isArray(lazy) || !lazy.every(item => typeof item === "string"))
+  ) {
+    throw new Error(`.lazy must be a boolean, array of strings, or a function`);
```

Comment:
Is the . at the beginning intentional?

Reply:
It's because it's a key in the configuration.

Reply:
I guess we don't really have a good naming scheme for these errors. I just included it since it is an object property. I can change it.

Reply:
At least this is consistent with the other error messages used in Babel but it's not super clear to me. What about: "the key 'lazy' in the Babel configuration must be a boolean". Can be done in a second PR. That reminds my that I still have a PR to add code frames to configuration issues. I can continue my work, what do you think?

Reply:
> What about: "the key 'lazy' in the Babel configuration must be a boolean". 

I don't think it's much clearer than what I have here, but I do think we could do better in general with work focused on making the error messages give better feedback.

> That reminds my that I still have a PR to add code frames to configuration issues. I can continue my work, what do you think?

Not sure I follow. What code frames could we show for this case? Do you mean showing information about the config file itself?